### PR TITLE
Normalize subfile path in H5 lib

### PR DESCRIPTION
**Pull request by nilsvu**
This has bothered me for a long time. A leading slash and _no_ extension for subfile names was required before, and failing to do this lead to rather useless error messages. Now, the leading slash and the extension are handled within the H5 lib, so it should be much easier to use.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/IO/H5/File.hpp
Comment (knelli2) [inline src/IO/H5/File.hpp:398]: Can you add a test of this normalization in the c++ tests?

**Review by knelli2 — CHANGES_REQUESTED** [review]
@nilsdeppe Could you take a quick look at this? I want to merge this before #5919 because it's used in there

File: src/IO/H5/File.hpp
Comment (nilsvu) [reply src/IO/H5/File.hpp:398]: done and squashed in

**Comment by nilsdeppe** [comment]
No need to wait for explicit approval from me.

**Review by knelli2 — APPROVED** [review]

**Pull request merged**
